### PR TITLE
fix(core): use correct host portal viewport for dropdown position

### PR DESCRIPTION
**Pull request by splincode**
## PR Type

What kind of change does this PR introduce?

- [x] Bugfix
- [ ] Feature
- [ ] Refactoring
- [ ] Build or CI related changes
- [ ] Tests related changes
- [ ] Documentation content changes

## What is the current behaviour?

Closes https://github.com/taiga-family/tui-editor/issues/201

image

## What is the new behaviour?

image

### CODE REVIEW

File: projects/core/directives/dropdown/dropdown-position.directive.ts [hunk 1]
```diff
@@ -44,15 +44,17 @@ export class TuiDropdownPositionDirective extends TuiPositionAccessor {
             top: hostRect.top - 2 * offset - viewport.top,
             bottom: viewport.bottom - hostRect.bottom - 2 * offset,
         } as const;
+        const left =
+            hostRect.left + width < viewport.right - offset ? hostRect.left : right;
         const position = {
             top: hostRect.top - offset - height,
             bottom: hostRect.bottom + offset,
-            right,
+            right: viewport.left > right ? viewport.left : right,
```
Comment (waterplea) [inline projects/core/directives/dropdown/dropdown-position.directive.ts:52]: Should it be `viewport.left + offset`? Also `Math.max(viewport.left + offset, right)`

File: projects/core/directives/dropdown/dropdown-position.directive.ts [hunk 2]
```diff
@@ -44,15 +44,17 @@ export class TuiDropdownPositionDirective extends TuiPositionAccessor {
...
             center:
                 hostRect.left + hostRect.width / 2 + width / 2 < viewport.right - offset
                     ? hostRect.left + hostRect.width / 2 - width / 2
                     : right,
-            left: hostRect.left + width < viewport.right - offset ? hostRect.left : right,
+            left: viewport.left > left ? viewport.left : left,
```
Comment (waterplea) [inline projects/core/directives/dropdown/dropdown-position.directive.ts:57]: Same here

File: projects/core/directives/dropdown/dropdown-position.directive.ts [hunk 3]
```diff
@@ -44,15 +44,17 @@ export class TuiDropdownPositionDirective extends TuiPositionAccessor {
             top: hostRect.top - 2 * offset - viewport.top,
             bottom: viewport.bottom - hostRect.bottom - 2 * offset,
         } as const;
+        const left =
+            hostRect.left + width < viewport.right - offset ? hostRect.left : right;
```
Comment (waterplea) [inline projects/core/directives/dropdown/dropdown-position.directive.ts:48]: I wonder if we can somehow simplify all the offset subtractions by just subtracting it from the viewport rect initially 🤔

**Review by waterplea — COMMENTED** [review]
Do we need something like that for hints too?

**Comment by splincode** [comment]
`Do we need something like that for hints too?` maybe separate PR

**Comment by splincode** [comment]
@vladimirpotekhin hello, please review me)

**Review by vladimirpotekhin — APPROVED** [review]

**Pull request merged**